pkg/odo/genericclioptions/context.go

```diff
@@ -471,12 +471,12 @@ func newDevfileContext(command *cobra.Command) *Context {
 		util.LogErrorAndExit(err, "")
 	}
 
+	internalCxt.EnvSpecificInfo = envInfo
```

Comment:
if we are setting the EnvSpecificInfo as part of devfile context initialisation then we don't need this https://github.com/openshift/odo/pull/3083/files#diff-ea37f2099454a7e283da6cc75ebe3285R67